pkg/service/model_build_load_balancer.go

```diff
@@ -109,20 +110,42 @@ func (t *defaultModelBuildTask) buildLoadBalancerTags(ctx context.Context) (map[
 	return t.buildAdditionalResourceTags(ctx)
 }
 
-func (t *defaultModelBuildTask) buildLoadBalancerSubnetMappings(_ context.Context, ec2Subnets []*ec2.Subnet) ([]elbv2model.SubnetMapping, error) {
+func (t *defaultModelBuildTask) buildLoadBalancerSubnetMappings(_ context.Context, scheme elbv2model.LoadBalancerScheme, ec2Subnets []*ec2.Subnet) ([]elbv2model.SubnetMapping, error) {
 	var eipAllocation []string
 	eipConfigured := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixEIPAllocations, &eipAllocation, t.service.Annotations)
-	if eipConfigured && len(eipAllocation) != len(ec2Subnets) {
-		return []elbv2model.SubnetMapping{}, errors.Errorf("number of EIP allocations (%d) and subnets (%d) must match", len(eipAllocation), len(ec2Subnets))
+	var privateIpv4Addresses []string
+	ipv4Configured := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixPrivateIpv4Addresses, &privateIpv4Addresses, t.service.Annotations)
+
+	// Validation
+	if eipConfigured && ipv4Configured {
+		return []elbv2model.SubnetMapping{}, errors.Errorf("Only one of EIP allocations and PrivateIpv4Addresses can be set")
+	}
+	if eipConfigured {
+		if scheme == elbv2model.LoadBalancerSchemeInternal {
+			return []elbv2model.SubnetMapping{}, errors.Errorf("EIP allocations can only be set for internet facing load balancers")
+		} else if len(eipAllocation) != len(ec2Subnets) {
+			return []elbv2model.SubnetMapping{}, errors.Errorf("number of EIP allocations (%d) and subnets (%d) must match", len(eipAllocation), len(ec2Subnets))
+		}
+	}
+	if ipv4Configured {
+		if scheme == elbv2model.LoadBalancerSchemeInternetFacing {
+			return []elbv2model.SubnetMapping{}, errors.Errorf("PrivateIpv4Addresses can only be set for internal balancers")
+		} else if len(privateIpv4Addresses) != len(ec2Subnets) {
+			return []elbv2model.SubnetMapping{}, errors.Errorf("number of PrivateIpv4Addresses (%d) and subnets (%d) must match", len(privateIpv4Addresses), len(ec2Subnets))
+		}
 	}
+
 	subnetMappings := make([]elbv2model.SubnetMapping, 0, len(ec2Subnets))
 	for idx, subnet := range ec2Subnets {
 		mapping := elbv2model.SubnetMapping{
 			SubnetID: aws.StringValue(subnet.SubnetId),
 		}
-		if idx < len(eipAllocation) {
+		if eipConfigured {
 			mapping.AllocationID = aws.String(eipAllocation[idx])
 		}
+		if ipv4Configured {
+			mapping.PrivateIPv4Address = aws.String(privateIpv4Addresses[idx])
```

Comment:
Since private address must be from the subnet CIDR range, lets map the IP addresses to the subnet CIDR range. That way the order of specification in the annotation doesn't matter.

Reply:
Added a method for matching the IP against the CIDR. Let me know what you think